recipe/meta.yaml

```diff
@@ -218,17 +221,19 @@ outputs:
         - pyarrow._hdfs
         # We can only test importing cuda package but cannot run when a
         # CUDA device is not available, for instance, when building from CI.
-        - pyarrow.cuda     # [cuda_compiler_version not in (undefined, "None")]
+        # On Windows, we cannot even test the import due to `nvcuda.dll` not
+        # being found, see conda-forge/conda-forge.github.io#1226
+        - pyarrow.cuda     # [cuda_compiler_version not in (undefined, "None") and not win]
```

Comment:
Can we check instead for the existence of the `pyarrow._cuda` Python module then? So we can at least ensure that it was built.

Reply:
What's the difference between `pyarrow._cuda` and `pyarrow.cuda`? If the former can be imported without needing `nvcuda.dll`, that would be an easy solution (and we could still test importing `pyarrow.cuda` on linux).

Reply:
`pyarrow.cuda` is a bit of Python boilerplate and `pyarrow._cuda` is the compiled Cython code linking to `libarrow_cuda`. Both need `nvcuda.dll` to work as `pyarrow.cuda` depends on `pyarrow._cuda`.

Reply:
OK, there's not much "glue" between `_cuda` and `cuda`: https://github.com/apache/arrow/blob/master/python/pyarrow/cuda.py

> @xhochy: Can we check instead for the existence of the `pyarrow._cuda` Python module then? So we can at least ensure that it was built.

I have to say I'm unable to identify which part of https://github.com/apache/arrow/blob/master/python/pyarrow/_cuda.pyx would trigger the DLL load (or is that added during compilation? shouldn't that by cython-only?) - but maybe it's possible to import a subset of:
```
from pyarrow._cuda import (Context, IpcMemHandle, CudaBuffer,
                           HostBuffer, BufferReader, BufferWriter,
                           new_host_buffer,
                           serialize_record_batch, read_message,
                           read_record_batch)
```

Reply:
Sorry, GH didn't show me your reply so I went searching myself. The part where I don't understand from the source of `_cuda.pyx` where the DLL-dependence comes in is still relevant though.